src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,288 @@
+                np.linspace(0, beat_times.size - 1, beat_times.size * steps_per_beat - 1)
+            )
+        return beat_steps_8th
+
+    def extrapolate_beat_times(self, beat_times, n_extend=1):
+        beat_times_function = scipy.interpolate.interp1d(
+            np.arange(beat_times.size),
+            beat_times,
+            bounds_error=False,
+            fill_value="extrapolate",
+        )
+        ext_beats = beat_times_function(np.linspace(0, beat_times.size + n_extend - 1, beat_times.size + n_extend))
+
+        return ext_beats
+
+    def preprocess_mel(
+        self,
+        audio,
+        beatstep,
+        n_bars,
+        padding_value,
+    ):
+        """Preprocessing for log-mel spectrogram"""
+
+        n_steps = n_bars * 4
+        n_target_step = len(beatstep)
+        ext_beatstep = self.extrapolate_beat_times(beatstep, (n_bars + 1) * 4 + 1)
+
+        batch = []
+        for i in range(0, n_target_step, n_steps):
+            start_idx = i
+            end_idx = min(i + n_steps, n_target_step)
+
+            start_sample = int(ext_beatstep[start_idx] * self.sampling_rate)
+            end_sample = int(ext_beatstep[end_idx] * self.sampling_rate)
+            feature = audio[start_sample:end_sample]
+            batch.append(feature)
+        batch = pad_sequence(batch, batch_first=True, padding_value=padding_value)
+
+        return batch, ext_beatstep
+
+    def single_preprocess(
+        self,
+        beatstep,
+        audio=None,
+    ):
+        """Preprocessing method for a single sequence."""
+
+        if audio is not None:
+            if len(audio.shape) != 1:
+                raise ValueError(
+                    f"Expected `audio` to be a single channel audio input of shape to be (n, ) but found {audio.shape}."
+                )
+
+        if beatstep[0] > 0.01:
+            logger.warning(f"`beatstep[0]` is not 0. All beatstep will be shifted by {beatstep[0]}.")
+            beatstep = beatstep - beatstep[0]
+
+        batch, ext_beatstep = self.preprocess_mel(
+            audio,
+            beatstep,
+            n_bars=self.n_bars,
+            padding_value=self.padding_value,
+        )
+
+        return batch, ext_beatstep
+
+    def __call__(
+        self,
+        raw_audio: Union[np.ndarray, List[float], List[np.ndarray]],
+        sampling_rate: int,
+        steps_per_beat: int = 2,
+        return_tensors: Optional[Union[str, TensorType]] = "pt",
+        **kwargs,
+    ) -> BatchFeature:
+        """
+        Main method to featurize and prepare for the model one sequence. Please note that `Pop2PianoFeatureExtractor`
+        only accepts one raw_audio at a time.
+
+        Args:
+            raw_audio (`np.ndarray`, `List`):
+                Denotes the raw_audio.
+            sampling_rate (`int`):
+                Denotes the Sampling Rate of `raw_audio`.
+            steps_per_beat (`int`, *optional*, defaults to 2):
+                Denotes Steps per beat.
```

Comment:
I don't think these descriptions are particularly useful, since they simply repeat the argument names. ;-)

```suggestion
            audio (`np.ndarray`, `List`):
                The sequence or batch of sequences to be processed. Each sequence can be a numpy array, a list of float
                values, a list of numpy arrays or a list of list of float values. 
            sampling_rate (`int`):
                The sampling rate at which the `audio` input was sampled. It is strongly recommended
                to pass `sampling_rate` at the forward call to prevent silent errors.
            steps_per_beat (`int`, *optional*, defaults to 2):
                An excellent description of what this argument actually means.
```